packages/store/addon/-private/ts-interfaces/minimum-serializer-interface.ts

```diff
@@ -0,0 +1,280 @@
+/**
+  ## Overview
+
+  In order to properly manage and present your data, `EmberData`
+  needs to understand the structure of data it receives.
+
+  `Serializers` convert data between the format used by an API
+  and the format `EmberData` understands.
+
+  Data received from an API response is `"normalized"` into 
+  [JSON:API](https://jsonapi.org/) (the format used internally
+  by `EmberData`), while data sent to an API is `"serialized"`
+  into the format the API expects.
+
+  ### Implementing a Serializer
+
+  There are only two required serializer methods, one for
+  normalizing data in `JSON:API`, and another for serializing
+  records via `Snapshot`s into the expected API.
+
+  To implement a serializer, export a class implementing the
+  [MinimumSerializerInterface](MinimumSerializerInterface) from
+  the `app/serializers/` directory. An example is below.
+
+  ```ts
+  import EmberObject from '@ember/object';
+
+  export default class ApplicationSerializer extends EmberObject {
+    normalizeResponse(_, __, rawPayload) {
+      return rawPayload;
+    }
+    serialize(snapshot, options) {
+      const serializedResource = {
+        id: snapshot.id(),
+        type: snapshot.modelName,
+        attributes: snapshot.attributes()
+      };
+
+      return serializedResource;
+    }
+  }
+ ```
+
+
+  #### Serializer Resolution
+
+  The instances of serializers defined in `app/serializers/` can be looked up
+  via `store.serializerFor(name)`.
+
+  It is recommended that applications define only a single `application` adapter and serializer
+  where possible.
+
+  `serializerFor` first attempts to find a serializer with an exact match on `name`,
+  then falls back to checking for the presence of a serializer named `application`.
+
+  Because most requests in `ember-data` occur from the perspective of a primary `type`
+  (or `modelName`) typically `serializerFor` will be used to find a serializer with a name
+  matching that of the primary resource `type` for the request, falling back to the `application`
+  serializer for those types that do not have a defined serializer. This is often described
+  as a `per-model` or `per-type` strategy for defining serializers. However, because APIs
+  rarely format payloads per-type but rather per-API-version, this may not be a desired strategy.
+
+  If you have multiple API formats and the per-type strategy is not viable, one strategy is to
+  write an `application` adapter and serializer that make use of `options` to specify the desired
+  format when making a request.
+
+  ### Using a Serializer
+
+  Any serializer in `app/serializers` can be looked up by `name` using `store.serializerFor(name)`.
+
+  ### Default Serializers
+
+  It is recommended that apps write their own serializer to best suit the needs of their API and
+  application.
+
+  However, for applications whose APIs are *very close to* or *exactly* the `REST` or `JSON:API`
```

Comment:
Do we need to say this specifically here? If their format is very close to what we expect, it is _equally_ easy to do the minimal interface and pass through (vs us JSONAPISerializer), right?

Reply:
Yes. I’m not sure the best way to word this for indicating that these are more optional convenience for getting started quickly than the best way to go.